### PR TITLE
Check for prototype bound name in class static accessors

### PR DESCRIPTION
**Pull request by dbatyai**

### CODE REVIEW

File: jerry-core/parser/js/js-parser-expr.c [hunk 1]
```diff
@@ -621,8 +621,16 @@ parser_parse_class_body (parser_context_t *context_p, /**< context */
       {
         is_computed = true;
       }
-      else if (!is_static && parser_is_constructor_literal (context_p))
+      else if (is_static)
       {
+        if (lexer_compare_literal_to_string (context_p, "prototype", 9))
```
Comment (rerobika) [inline jerry-core/parser/js/js-parser-expr.c:626]: Can we ensure that the current literal is ident or string?

Also, as I see we should also throw error in the following case :
```js
class A {
  static get prototyp\u{0065}() {}
}
```

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/parser/js/js-lexer.c [hunk 2]
```diff
@@ -3707,12 +3707,12 @@ lexer_compare_literal_to_string (parser_context_t *context_p, /**< context */
                                  const char *string_p, /**< string */
                                  size_t string_length) /**< string length */
 {
-  JERRY_ASSERT (context_p->token.type == LEXER_LITERAL
-                && (context_p->token.lit_location.type == LEXER_IDENT_LITERAL
-                    || context_p->token.lit_location.type == LEXER_STRING_LITERAL));
+  JERRY_ASSERT (context_p->token.type == LEXER_LITERAL);
```
Comment (zherczeg) [inline jerry-core/parser/js/js-lexer.c:3710]: I think we should not change this function.

**Comment by rerobika** [comment]
LGTM

**Review by rerobika — APPROVED** [review]
LGTM

File: tests/jerry/es.next/class.js [hunk 3]
```diff
@@ -49,6 +49,11 @@ must_throw("class class {}");
 must_throw("class A { constructor() {} this.a = 5 }");
 must_throw("class A { constructor() {} constructor() {} }");
 must_throw("class A { static prototype() {} }");
+must_throw("class A { static get prototype() {} }");
+must_throw("class A { static set prototype() {} }");
```
Comment (zherczeg) [inline tests/jerry/es.next/class.js:53]: I would add some testcases for the string variant as well:
`must_throw("class A { static get 'prototype'() {} }");`

**Review by zherczeg — CHANGES_REQUESTED** [review]

**Review by zherczeg — APPROVED** [review]
LGTM

**Comment by zherczeg** [comment]
Please resolve conflicts first, then you can land the patch.

**Pull request merged**